File: csharp/kindergarten-garden/KindergartenGarden.cs
using System;
using System.Collections.Generic;
using System.Linq;

public enum Plant
{
    Violets = 'V',
    Radishes = 'R',
    Clover = 'C',
    Grass = 'G'
}

public class KindergartenGarden
{
    IEnumerable<Plant> Garden;
    string[] Students = { "Alice", "Bob", "Charlie", "David", "Eve", "Fred", "Ginny", "Harriet", "Ileana", "Joseph", "Kincaid", "Larry" };

    public KindergartenGarden(string diagram) =>
        Garden = diagram.Where(c => Enum.IsDefined(typeof(Plant), (int)c)).Select(c => (Plant)c);

    public IEnumerable<Plant> Plants(string student)
    {
        int n = 2 * Array.IndexOf(Students, student);
        return Garden.Skip(n).Take(2).Concat(Garden.Skip(n + Garden.Count() / 2).Take(2));
    }
}

File: fsharp/kindergarten-garden/KindergartenGarden.fs
module KindergartenGarden

type Plant = Violets | Grass | Radishes | Clover

let private getPlant = function
    | 'V' -> Some Violets
    | 'G' -> Some Grass
    | 'R' -> Some Radishes
    | 'C' -> Some Clover
    | _ -> None

let private students =
    "Alice, Bob, Charlie, David, Eve, Fred, Ginny, Harriet, Ileana, Joseph, Kincaid, Larry".Split ", "
    |> Array.mapi (fun index name -> (name, index))
    |> dict

let plants (diagram: string) student =
    diagram.Split '\n'
    |> Seq.collect (Seq.skip (2 * students[student]) >> Seq.take 2)
    |> Seq.choose getPlant
    |> List.ofSeq
